apps/artemis-web/middleware/limiter.ts

```diff
@@ -0,0 +1,57 @@
+import rateLimit from "express-rate-limit";
+require('dotenv').config();
+
+
+function calculateNextResetTime(windowMs) {
+    const d = new Date();
+    d.setMilliseconds(d.getMilliseconds() + windowMs);
+    return d;
+}
+
+function MemoryStore(windowMs, file) {
+    let hits = {};
+    let resetTime = calculateNextResetTime(windowMs);
+
+    this.incr = function (key, cb) {
+        if (hits[file + "," + key]) {
+            hits[file + "," + key]++;
+        } else {
+            hits[file + "," + key] = 1;
+        }
+
+        cb(null, hits[file + "," + key], resetTime);
+    };
+
+    this.decrement = function (key) {
+        if (hits[file + "," + key]) {
+            hits[file + "," + key]--;
+        }
+    };
+
+    // export an API to allow hits all IPs to be reset
+    this.resetAll = function () {
+        hits = {};
+        resetTime = calculateNextResetTime(windowMs);
+    };
+
+    // export an API to allow hits from one IP to be reset
+    this.resetKey = function (key) {
+        delete hits[file + "," + key];
+    };
+
+    // simply reset ALL hits every windowMs
+    const interval = setInterval(this.resetAll, windowMs);
+    if (interval.unref) {
+        interval.unref();
+    }
+}
+
+const limiter = (file) => {
+    return rateLimit({
+        windowMs: parseInt(process.env.LIMIT_WINDOW),
```

Comment:
important question: what happens if the `LIMIT_WINDOW` or `LIMIT_REQUESTS` env vars are not defined? We should have some default values in in case the environment is not complete (this can happen for example if we merge the code before there is support in the ARTEMIS mono-repo with the env vars). Please fix here so that this does not fail and I will in parallel add the env vars to the mono-repo which includes the environment and the backend.

Reply:
I hardcoded a default value now :)